molecularnodes/nodes/styles.py

```diff
@@ -0,0 +1,281 @@
+"""
+Style Classes
+
+MolecularNodes uses Geometry Nodes to define molecule representations within blender. However,
+it is desireable to make these nodes accessible for scripting. The classes defined in this file
+provide a way to script against the GeometryNode nodetrees. When invoked on a style these classes
+will simply override the base styles.
+
+Because the nodetrees in the MN_data_file_{version}.blend are the source of truth, we add
+function to parity between the nodetrees and the class representations.
+
+Each style uses a Socket dataclass to define the mapping between class attributes and
+Blender node inputs, making it easier to maintain and extend the styles. The Socket
+dataclass establishes a clear relationship between Python attribute names and the corresponding
+Blender node input socket names, providing type safety and better IDE support than the
+previous dictionary-based approach.
+"""
+
+from dataclasses import dataclass
+from typing import List, Optional, Tuple
+from bpy.types import GeometryNodeGroup
+
+__all__ = [
+    "StyleBallandStick",
+    "StyleCartoon",
+    "StyleRibbon",
+    "StyleSpheres",
+    "StyleSticks",
+    "StyleSurface",
+    "StyleBase",
+]
+
+
+@dataclass
+class Socket:
+    """Represents a mapping between a class attribute and a Blender node input socket.
+
+    Attributes:
+        name: The name of the attribute in the Style class
+        blendername: The corresponding name of the input socket in the Blender node
+    """
+
+    name: str
+    blendername: Optional[str] = None
+
+
+SocketInfo = List[Socket]
+
+
+class StyleBase:
+    style: str
+    portdata: SocketInfo = []
+
+    def update_style_node(self, node_style: GeometryNodeGroup):
+        """Update the Blender node inputs with values from this style's attributes.
+
+        Args:
+            node_style: The Blender GeometryNodeGroup to update
+        """
+        for input in node_style.inputs:
+            if input.type != "GEOMETRY":
+                for arg in self.portdata:
+                    if input.name == arg.blendername:
+                        input.default_value = getattr(self, arg.name)
+
+
+class StyleBallandStick(StyleBase):
+    style = "ball_and_stick"
+    portdata: SocketInfo = [
```

Comment:
rename from port to socket

Reply:
this should be renamed very time it shows up

Reply:
done